src/ninshiki_cpp/utils/circle.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2021 ICHIRO ITS
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "ninshiki_cpp/utils/circle.hpp"
+
+namespace ninshiki_cpp::utils
+{
+
+Circle::Circle(const std::vector<cv::Point> & contour)
+: center(cv::Point2f(-1, -1)), radius(0.0)
+{
+    cv::minEnclosingCircle(cv::Mat(contour), center, radius);
+}
+
+cv::Mat Circle::get_binary_mat_line(const cv::Size & mat_size, int line_size)
+{
+    cv::Mat binary_mat(mat_size, CV_8UC1);
+    binary_mat = cv::Scalar(0);
+
+    cv::circle(binary_mat, center, radius, 255, line_size);
+
+    return binary_mat;
+}
```

Comment:
I'm not sure how are you going to call this, but if there's multiple circles and we want to draw all of them, each will create a new instance of an image object. To avoid that, we can modify this function to draw into an existing image, somehow like this:

```suggestion
cv::Mat Circle::draw(cv::Mat image, int line_size)
{
    cv::circle(image, center, radius, 255, line_size);
}
```